Fix/forms/setinputgrouplabelasspan

dagfrode opened the pull request:
<!-- Gi saken en oppsummerende tittel ovenfor. -->

## Beskrivelse

Teamene får enda `setinputgrouplabelasspan` siden de bruker 

```tsx

<InputGroup>
{(inputProps) => (
    <>
        <Input
            value={formatAmountV2(estateValue)}
            name="estateValue"
            {...inputProps}
        />
    </>
)}
</InputGroup>
```

fremfor

```tsx
<InputGroup>
        <Input
            value={formatAmountV2(estateValue)}
            name="estateValue"
        />
</InputGroup>
```

Ikke sikker på hvorfor de gjør det, men de gjør det.

Fjerner derfor hacken. Det gjør den muligens litt mindre UU vennlig, men fjerner warningen og er ikke nødvendig for testene lenger heller pga egne testhelpers.

Fixes [#2738](https://github.com/SpareBank1/designsystem/issues/2738)

Oppdaterer og testhelper slik at den støtter å sende inn tomme eller delvis tomme datoer og on blur etter year slik at hvis en sender inn en delvis dato blir den satt riktig.

Gjør og et par små endringer for å få le date ting å fungere riktig igjen etter at vi gjorde datepicker om til konntrollert komponent.

Gikk fra å være en liten quick fix til flere problemer betaling oppllevde her.

dagfrode commented on packages/ffe-datepicker-react/src/datepicker/DatepickerComp.tsx:
```diff
@@ -117,16 +117,17 @@ export const DatepickerComp: React.FC<DatepickerCompProps> = ({
-            ) {
-                date.adjust({ period: 'Y', offset: -100 });
-            }
+            // Kan vi fjerne dette?
```
Fikk spørsmål om det var meningen at datoen skulle komplete til `1928` og ikke `2028` når de skrev inn `28`. De har nok satt en max date. Men da er jo det noe de validerer på selv. Gir mening at de ikke får velge i kalenderen, men rart å endre det brukeren skriver inn.

hagenek replied:
Det er vel bare å fjerne? Rar oppførsel imo.

tuva-odegard replied:
Ikke bare å fjerne, vi ble bedt om å legge det inn igjen etter siste oppdatering i november! Handler spesielt om hvis man kopierer inn datoer som ikke har fullstendig år. Usikker på logikken om når det blir 2028 og 1928, men det burde sees på hvis den oppfører seg unormalt. Kanskje vi burde se på en bedre logikk om når det burde bli 19 og når det blir 20.

dagfrode replied:
aight, da lar jeg det være og event lager en egen issue på den casen :)

tuva-odegard replied:
Når jeg tester nå så fungerer det med å autocomplete til 20- i alle fall. MaxDate fungerer ikke å putte inn på storybook, så får ikke testet om maxdate påvirker om den velger 19 eller 29. Usikker på hva denne koden gjør i så fall, men ikke fan av å ha kommentert ut kode liggende. Så heller fjern den hvis vi ikke skal bruke den!

dagfrode replied:
Enig. Kommenterte det ut for å diskutere det. Jeg legger det tilbake igjen

hagenek reviewed (dismissed):
Gode forbedringer.

tuva-odegard reviewed (changes requested):
Burde ikke være 3 forskjellige commits på 1 pakke kanskje?

dagfrode commented:
> Burde ikke være 3 forskjellige commits på 1 pakke kanskje?

Skal ta å rett opp i committene

dagfrode commented:
feil knapp :sweat_smile:

tuva-odegard commented on packages/ffe-datepicker-react/src/datepicker/DatepickerInContext.spec.tsx:
```diff
@@ -45,4 +46,40 @@ describe('<InputGroup><Datepicker /></InputGroup>', () => {
+        expect(datepicker.getValue()).toStrictEqual('02.02.2128');
+    });
+
+    it('no year turns into current year', async () => {
```
Nice!

tuva-odegard commented on packages/ffe-datepicker-react/src/datepicker/DatepickerInContext.spec.tsx:
```diff
@@ -45,4 +46,40 @@ describe('<InputGroup><Datepicker /></InputGroup>', () => {
         expect(datepicker.getValue()).toStrictEqual('06.05.2024');
     });
+
+    it('datepicker can be updated by testing functions to empty', async () => {
```
Skjønner ikke helt hva denne prøver å si, hva betyr functions to empty? Kunne det vært mer beskrivende?

tuva-odegard commented on packages/ffe-datepicker-react/src/datepicker/DatepickerInContext.spec.tsx:
```diff
@@ -45,4 +46,40 @@ describe('<InputGroup><Datepicker /></InputGroup>', () => {
+        expect(datepicker.getValue()).toBeNull();
+    });
+
+    it('2 number year date works', async () => {
```
"Works" 😅 Burde kanskje beskrive hva "works" er?

dagfrode committed:
fix(ffe-form-react): Remove setInputGroupLabelAsSpan

tuva-odegard commented on packages/ffe-datepicker-react/src/datepicker/DatepickerInContext.spec.tsx:
```diff
@@ -45,4 +46,40 @@ describe('<InputGroup><Datepicker /></InputGroup>', () => {
+        const datepicker = getDatepickerByLabelText('Datovelger');
+        await datepicker.setValue('02.02.');
+
+        expect(datepicker.getValue()).toStrictEqual('02.02.2025');
```
ahhh så ikke denne før nå. Sorry!! Men litt kjipt om testene feiler 01.01 neste år. new Date().getYear?

dagfrode replied:
No wrorries, er jeg som skal beklage, er jeg som var lat

dagfrode replied:
Oppdatert

dagfrode committed:
fix(ffe-datepicker-react): Remove setInputGroupLabelAsSpan and fix Testhelper

- Fix testhelper with blur and partial year
- Fix testhelper to work with empty date

tuva-odegard commented:
.... ahhhh trodde jeg testet alt i stad. Men calender og dateinput er ikke synca på år-fallbacken tydeligvis.....
![image](https://github.com/user-attachments/assets/95801bc1-a06b-402b-8ebb-9a8feabb0128)

dagfrode commented:
> ahhhh trodde jeg testet alt i stad. Men calender og dateinput er ikke synca på år-fallbacken tydeligvis.....

![](https://media.giphy.com/media/wLG70BswTt0Nq/giphy.gif?cid=790b7611fzea1dl8hz9dpnyndrz67jbwdy97fv9avvjwzjeh&ep=v1_gifs_search&rid=giphy.gif&ct=g)

Hade jeg ikke engang tenkt på å teste. Hørs ut som et mandagsproblem :sweat_smile:

dagfrode commented:
> .... ahhhh trodde jeg testet alt i stad. Men calender og dateinput er ikke synca på år-fallbacken tydeligvis.....

Trakk dette ut i en egen feil da den er der i dag: https://github.com/SpareBank1/designsystem/issues/2753
Så på den i 15 min, ble ikke noe klokere. Vil få inn denne så betling kommer seg videre.
